python/tank/bootstrap/manager.py

```diff
@@ -680,7 +689,7 @@ def _bootstrap_sgtk(self, engine_name, entity, progress_callback=None):
         self._report_progress(progress_callback, self._STARTING_TOOLKIT_RATE, "Starting up Toolkit...")
         tk = config.get_tk_instance(self._sg_user)
 
-        if not isinstance(config, InstalledConfiguration):
+        if not config.has_local_bundle_cache:
```

Comment:
I understand the appeal. However, as it currently is the progress reporting is entirely handled at the ToolkitManager level. In order to do what you are suggesting, we would have to pass down the progress callback to the configuration, which I feel is at odds with the current implementation at the moment.

What do you think?